frontend/src/components/Common/DevelopmentalStageFilter.jsx

```diff
@@ -87,6 +87,7 @@ const DevelopmentalStageFilter = () => {
         valueLabelDisplay="auto"
         marks={marks}
         aria-label="Developmental Stages Filter"
+        onChange={setSearching}
```

Comment:
It shows the loading spinner but it does so even if the value didn't change, need here a comparison so that it triggers only if previous value changed.